src/Sarif/NotYetAutoGenerated/Run.cs

```diff
@@ -0,0 +1,447 @@
+        public string BaselineInstanceGuid { get; set; }
+
+        /// <summary>
+        /// The MIME type of all rich text message properties in the run. Default: "text/markdown;variant=GFM"
+        /// </summary>
+        [DataMember(Name = "richMessageMimeType", IsRequired = false, EmitDefaultValue = false)]
+        [DefaultValue("text/markdown;variant=GFM")]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public string RichMessageMimeType { get; set; }
+
+        /// <summary>
+        /// The string used to replace sensitive information in a redaction-aware property.
+        /// </summary>
+        [DataMember(Name = "redactionToken", IsRequired = false, EmitDefaultValue = false)]
+        public string RedactionToken { get; set; }
+
+        /// <summary>
+        /// Specifies the default encoding for any file object that refers to a text file.
+        /// </summary>
+        [DataMember(Name = "defaultFileEncoding", IsRequired = false, EmitDefaultValue = false)]
+        public string DefaultFileEncoding { get; set; }
+
+        /// <summary>
+        /// An ordered list of character sequences that were treated as line breaks when computing region information for the run.
+        /// </summary>
+        [DataMember(Name = "newlineSequences", IsRequired = false, EmitDefaultValue = false)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public IList<string> NewlineSequences { get; set; }
+
+        /// <summary>
+        /// Specifies the unit in which the tool measures columns.
+        /// </summary>
+        [DataMember(Name = "columnKind", IsRequired = false, EmitDefaultValue = false)]
+        [DefaultValue(ColumnKind.UnicodeCodePoints)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        [JsonConverter(typeof(Microsoft.CodeAnalysis.Sarif.Readers.EnumConverter))]
+        public ColumnKind ColumnKind { get; set; }
+
+        /// <summary>
+        /// References to external property files that should be inlined with the content of a root log file.
+        /// </summary>
+        [DataMember(Name = "externalPropertyFiles", IsRequired = false, EmitDefaultValue = false)]
+        public ExternalPropertyFiles ExternalPropertyFiles { get; set; }
+
+        /// <summary>
+        /// Key/value pairs that provide additional information about the run.
+        /// </summary>
+        [DataMember(Name = "properties", IsRequired = false, EmitDefaultValue = false)]
+        internal override IDictionary<string, SerializedPropertyInfo> Properties { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Run" /> class.
+        /// </summary>
+        public Run()
+        {
+            RichMessageMimeType = "text/markdown;variant=GFM";
+            ColumnKind = ColumnKind.Utf16CodeUnits;
```

Comment:
>ColumnKind.Utf16CodeUnits [](start = 25, length = 25)

Here's your column kind issue. For us to maintain reasonable compatibility, we need to instantiate this type by default with Utf16CodeUnits (which is not the literal default given by the schema, instead it's one that imposed by our Windows SDK to dig a 'pit of success' for this platform. 

By doing this, however, we also need to make sure to strip the other default handling. Otherwise we would never round-trip the alternate value (and actual default UnicodeCodePoints). That bad scenario would look like this:

1) user creates a log file, we init to utf16codeunits. user explicitly resets to unicodecodepoints.
2) user persist log file, and unicodecodepoints is elided (because it is the default)
3) user reopens log file, ctor inits to utf16codeunits and the unicodecodepoints setting is lost.

Instead, we basically overwrite everything such that run.columnKind is always persisted (by our SDK). If we happen to open a file without this value set, our SDK will treat it as Utf16CodeUnits.

Reply:
Having a knob to avoid emitting a schema-specified default would be helpful but is an incomplete solution, because we follow up with this by providing an alternate OM initialization only default. Not sure how to express this subtlety in jschema or if it even warrants doing so.

Note that all this would be much easier if this platform-specific data was required. We rejected doing this, however, because it seemed unnecessarily burdensome for tools that don't emit column details.

---
In reply to: [246527802](https://github.com/Microsoft/sarif-sdk/pull/1204#discussion_r246527802) [](ancestors = 246527802,246159777)

Reply:
Another option is to have the hint supply the overridden value.

---
In reply to: [246546734](https://github.com/Microsoft/sarif-sdk/pull/1204#discussion_r246546734) [](ancestors = 246546734,246527802,246159777)